Mixins

AlbertoElias opened the pull request:
Refactors to mixins, fixes #37 and #38, changes cross to use multiplication character.

kaelig commented:
Can you update the gitignore file and add `docs/` to it?

AlbertoElias commented:
@kaelig done :)

kaelig commented on src/scss/_arrow.scss:
```diff
@@ -1,4 +1,14 @@
+/// @access private
+///
+/// @param {Number} size
+///
```
Unnecessary comment line

kaelig commented on src/scss/_arrow.scss:
```diff
@@ -17,7 +27,12 @@
+/// @access private
+///
+/// @param {Number} size
+///
```
Unnecessary comment line

kaelig commented on src/scss/_arrow.scss:
```diff
@@ -34,6 +49,10 @@
 
+/// Top arrow styling
+///
+/// @param {Color} color
```
The default color `#74736c` is `pink-tint5`, should it be requiring o-colors instead of having it hard coded?

kaelig replied:
Also, default values should be supplied as such:

`@param {type} $name (default value) - description`

In that case not sure how it would be written if the default requires o-colors.

kaelig commented on src/scss/_arrow.scss:
```diff
@@ -64,7 +87,12 @@
+///
+/// @access private
+///
+/// @param {Number} size
```
Please supply default value in the doc

kaelig commented on src/scss/_compact.scss:
```diff
@@ -0,0 +1,44 @@
+/// @requires {mixin} oFtTypographyFontSize
+///
+@mixin oOverlayCompactHeading {
+	@include oFtTypographyFontSize(12);
```
This equals to `font-size: 12px`, not sure we need to depend on this mixin here

kaelig commented on src/scss/_fullscreen.scss:
```diff
@@ -0,0 +1,38 @@
+///
+/// @param {String} fill - 'width' or 'height'
+///
+@mixin oOverlayFullscreenContent($fill) {
```
Writing a mixin to output either `width: 100%` or `height: 100%` seems overkill.

If you think it absolutely makes sense to have it, could be written in one line:

```scss
#{$fill}: 100%;
```

kaelig commented on src/scss/_shaded.scss:
```diff
@@ -0,0 +1,25 @@
+
+///
+/// Add to the close icon for a shaded overlay
+///
```
Missing `@requires`

kaelig commented on src/scss/_shaded.scss:
```diff
@@ -0,0 +1,25 @@
+
+///
+/// Add to the header for a shaded overlay
+///
```
Missing `@requires`

kaelig commented on src/scss/_shaded.scss:
```diff
@@ -0,0 +1,25 @@
+@mixin oOverlayHeadingShadedCloseIcon {
+	@include oColorsFor(o-overlay-close-shaded);
+
+	#{$o-hoverable-if-hover-enabled} &:hover {
```
It would be a good time to add the `:focus` state too.

kaelig commented on src/scss/overlay.scss:
```diff
@@ -25,23 +40,35 @@
+/// @requires {function} oColorsGetColorFor
+///
+@mixin oOverlayHeading {
 	@include oFtTypographyFontSize(16);
```
Same as `font-size: 16px;`

kaelig commented on src/scss/overlay.scss:
```diff
@@ -51,39 +78,36 @@
 
+	&:after {
+		// Hex code for multiplication sign
+		content: "\00D7";
```
`×` would avoid having to comment that.

AlbertoElias commented:
@kaelig I've done those fixes

kaelig commented:
I can't see the `@requires` in some functions. Is that normal?

Some SassDoc comments start and finish with `///` (empty line). Not sure of the impact on SassDoc parsing so it might be fine.

AlbertoElias commented:
I added the ones you mentioned. Just checked and found a couple more I was missing.

Ah, I thought I only had to remove the finishing `///`, I've removed the starting ones as well

kaelig commented:
I'll let you choose one notation between `@requires` and `@require` (with and without a trailing `s`).

The preferred one is `@requires` but I've been guilty of using both.

kaelig commented on src/scss/_arrow.scss:
```diff
@@ -34,7 +49,12 @@
+/// @requires {mixin} _oOverlayVerticalArrow
+@mixin oOverlayUpArrow($color: oColorsGetColorFor(o-overlay, border)) {
 	@include _oOverlayVerticalArrow();
 	&:before, &:after {
```
One selector per line

kaelig commented on src/scss/_arrow.scss:
```diff
@@ -49,7 +69,12 @@
+/// @requires {mixin} _oOverlayVerticalArrow
+@mixin oOverlayDownArrow($color: oColorsGetColorFor(o-overlay, border)) {
 	@include _oOverlayVerticalArrow();
 	&:before, &:after {
```
One selector per line

kaelig commented on src/scss/_arrow.scss:
```diff
@@ -64,7 +89,13 @@
+/// @requires {mixin} _oOverlayArrow
 @mixin _oOverlayHorizontalArrow($size: 10px) {
 	@include _oOverlayArrow();
 	&:before, &:after {
```
One selector per line

kaelig commented on src/scss/_arrow.scss:
```diff
@@ -81,7 +112,12 @@
+/// @requires {mixin} _oOverlayHorizontalArrow
+@mixin oOverlayLeftArrow($color: oColorsGetColorFor(o-overlay, border)) {
 	@include _oOverlayHorizontalArrow();
 	&:before, &:after {
```
Ditto (and below as well)

AlbertoElias commented:
@kaelig done. Chose requires

kaelig commented on src/scss/_arrow.scss:
```diff
@@ -96,7 +132,12 @@
+/// @param {Color} color (oColorsGetColorFor(o-overlay, border))
+///
+/// @requires {mixin} _oOverlayHorizontalArrow
+@mixin oOverlayRightArrow($color: oColorsGetColorFor(o-overlay, border)) {
```
Should `oColorsGetColorFor(o-overlay, border)` be assigned to a global variable setting for the whole module (since it's used in multiple places)?

AlbertoElias replied:
I think it being a color use case is enough of an abstraction

kaelig replied:
Oh yeah true

kaelig commented:
:+1: :facepunch:
